# Конспект семинара 28

Цахилова Тома

## Тест Пирсона

### Задача:

Вовочка ловит покемонов. Всего он поймал 60 штук:

* 15 Бульбазавров

* 30 Чармандеров

* 15 Иви


![](http://gogopokemon.ru/wp-content/uploads/2016/07/2438704-1202149925_t.png)
![](https://upload.wikimedia.org/wikipedia/ru/4/49/%D0%9F%D0%BE%D0%BA%D0%B5%D0%BC%D0%BE%D0%BD_%D0%98%D0%B2%D0%B8.png)
![](https://i.ytimg.com/vi/w1klh6l_zSo/hqdefault.jpg)

$H_0$ Бульбазавры, Чармандеры и Иви встречаются равновероятно

$H_a$ Бульбазавры, Чармандеры и Иви встречаются **не** равновероятно

Уровень значимости $\alpha = 10\%$

$$ Q = \dfrac{\sum\limits_{i=1}^n\nu_i-np_i}{np_i} \sim _{H_0} \chi^2_{s-k-1} = \chi^2_2 $$

$$s = 3, k = 0$$


$$ Q = \frac{(15-20)^2 + (30-20)^2 +(15-20)^2}{20}=\frac{25+100+25}{20}=7.5$$

### Лайфхак, как решить, какой хвост нам нужен:

* игнорируем наши данные
* верим в $H_0$, хотя на самом деле верна $H_a$ 
* что будет со статистикой?
* $n \rightarrow \infty \Rightarrow Q \rightarrow \infty$, т.е., большие значения статистики говорят в пользу отклонения гипотезы

Работает с любыми статистиками - логику включать очень полезно!

Функция плотности хи-квадрата (нам нужна зелененькая линия):

![](https://upload.wikimedia.org/wikipedia/commons/thumb/3/35/Chi-square_pdf.svg/321px-Chi-square_pdf.svg.png)

$P(0\le Q\le 4.6) = 0.9$

$Q_{cr} = 4.61 < 7.5 \Rightarrow$ гипотеза $H_0$ отвергается

## LR

Решим ту же задачу с помощью LR

$$LR = 2 ( \ln L(\hat{\theta}_{ML}) - \ln L(\hat{\theta}_{H0})\ ) \sim \chi^2_q  $$

где q - число линейно независимых уравнений, задающих $H_0$

(в нашем случае q=2)

### Лайфхак, как определить q

#### Когда тестируем гипотезу:

* есть тестируемая часть ( $\mu = 1\ или \ \sigma^2 =42)$ - отсюда берется q
* есть нетестируемые предположения $\left( \sum\limits_{i=1}^n p_i = 1 \right) $

Составим функцию прадоподобия:

$$L = P(v_1=15, v_2=30,v_3=15) = C_{60}^{15} \ p_1^{15}\ C_{45}^{30}\ p_2^{30}(1-p_1-p_2)^{15} $$

$$\ln{L} = \ln C_{60}^{15} + \ln C_{45}^{30} + 15\ln p_1 +30\ln p_2 +15\ln{(1-p_1-p_2)}$$


$$\begin{cases}
\dfrac{\partial \ln L}{\partial p_1} = 15\dfrac{1}{p_1} - 15\dfrac{1}{1-p_1-p_2}\\
\dfrac{\partial \ln L}{\partial p_2} = 30\dfrac{1}{p_2} - 15\dfrac{1}{1-p_1-p_2}
\end{cases}$$

$$\begin{cases}
15\dfrac{1}{\widehat{p_1}} - 15\dfrac{1}{1-\widehat{p_1}-\widehat{p_2}}=0\\
30\dfrac{1}{\widehat{p_2}} - 15\dfrac{1}{1-\widehat{p_1}-\widehat{p_2}}=0
\end{cases}$$

Решив эту несложную систему уравнений или воспользовавшись методом 9-ти классника, получим, что:

$$
\widehat{p_1}_{ML}=\widehat{p_3}_{ML}=\frac{15}{60}=\frac{1}{4} \\
\widehat{p_3}_{ML}=\frac{1}{2}
$$

Посчитаем значение LR!

$$
LR = 2 \left(\ln C_{60}^{15} + \ln C_{45}^{30} + 15\ln\frac{1}{4}  +30\ln\frac{1}{2} +15\ln\frac{1}{4} - \ln C_{60}^{15} - \ln C_{45}^{30} - 15\ln\frac{1}{3}  -30\ln\frac{1}{3} -15\ln\frac{1}{3}\right) = 2(-90\ln2 +60\ln3)
$$

In [4]:
import numpy as np

In [10]:
LR = 2*(-90*np.log(2)+60*np.log(3))
LR

7.0669821393830148

Как мы помним, $\chi^2_{cr} = 4.61 < 7.07 \Rightarrow$ гипотеза $H_0$ отвергается

#### Пользуемся нашим лайфхаком №1 (логикой) для определения того, какой хвост нужен:

При неправдоподобной $H_0$ значение функции правдоподобия будет намного меньше, чем значение в точке, полученной через ML, т.е. $LR \rightarrow \infty$

Значит, отвергаем гипотезу при больших значениях LR (правый хвост)

**Ремарка 1** $LR \ge 0$ _**всегда**_ (т.к. из максимального значения логарифма функции правдоподобия вычитаем какое-то другое.

**Ремарка 2** LR, ML и тест Вальда применимы _**всегда**_!



![](http://lurkmore.so/images/thumb/7/78/Slowpoke.svg/200px-Slowpoke.svg.png)

Полезненькое:

[Чит-шит по makdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)

[Хлеб да соль картинки с Лукашенко](http://pikabu.ru/story/bulbazavr_ya_vyibirayu_tebya_2760931)

[Все сезоны Покемонов](http://onlinemultfilmy.ru/pokemon/)